serializers/src/main/java/io/pravega/schemaregistry/serializers/ProtobufDeserializer.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import com.google.common.base.Preconditions;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.pravega.schemaregistry.client.SchemaRegistryClient;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.schemas.ProtobufSchema;
+
+import java.io.InputStream;
+
+public class ProtobufDeserializer<T extends GeneratedMessageV3> extends AbstractDeserializer<T> {
+    private final ProtobufSchema<T> protobufSchema;
+    ProtobufDeserializer(String groupId, SchemaRegistryClient client,
+                         ProtobufSchema<T> schema, SerializerConfig.Decoders decoder,
+                         EncodingCache encodingCache, boolean encodeHeader) {
+        super(groupId, client, schema, true, decoder, encodingCache, encodeHeader);
+        Preconditions.checkNotNull(schema);
+        this.protobufSchema = schema;
+    }
+
+    @Override
+    protected T deserialize(InputStream inputStream, SchemaInfo writerSchemaInfo, SchemaInfo readerSchemaInfo) {
+        try {
+            return protobufSchema.getParser().parseFrom(inputStream);
+        } catch (InvalidProtocolBufferException e) {
+            throw new IllegalArgumentException("Invalid bytes", e);
```

Comment:
Should an IOException be thrown here too? JsonDeserializer throws an IOExcpetion incase of  and error processing . `JsonProcessingException extends java.io.IOException`

Reply:
This is an invalid protocol buffer. So illegal argument is appropriate as the input to this method is invalid.

Reply:
In the case of `JSONDeserializer` invalid data causes the `io.pravega.schemaregistry.serializers.AbstractDeserializer#deserialize(...) throws IOException` method to throw JsonParseException , where `JsonParseException` extends `java.io.IOException`. We can keep the same pattern here for Protobuf too.